### PR TITLE
Auto-generated documentation using Sphinx

### PR DESCRIPTION
**Pull request by Prakhar0409**
## Purpose
Generate documentation from docstrings #69 

## Approach
Used Sphinx (automodule) to generate the documentation. Generated html and latex forms. 

## New Issues/FIXME?
Hosting the HTML docs using some mechanism like github pages

## Learning
Use sphinx for python3 since it would need to go through your code while generating documentation (and might give error if you are using sphinx for python2)

### CODE REVIEW

**Comment by Prakhar0409** [comment]
Oops. Thanks for your comments. I have updated the PR:)

File: README.md [hunk 1]
```diff
@@ -83,6 +83,10 @@ You can generate a JSON file like this (each element is an user and this propert
 ```
 
 
+## Documentation
+
+Documentation can be generated by moving into the docs folder and running `make html`
```
Comment (iblancasa) [inline README.md]: When I follow these steps, the output is:
```bash
make: *** No rule to make target 'html'.  Stop.
```

**Comment by Prakhar0409** [comment]
Sorry for being careless. Had messed up the Makefile while trying to generate pdf docs. I have corrected the error.

**Comment by iblancasa** [comment]
This is the output when I run `make html`:
```bash
make html
Running Sphinx v1.6.4
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 3 source files that are out of date
updating environment: 3 added, 0 changed, 0 removed
reading sources... [100%] install                                                                                                        
WARNING: /home/israel/projects/fork/docs/code.rst:4: (WARNING/2) autodoc: failed to import module 'ghcity'; the following exception was raised:
Traceback (most recent call last):
  File "/home/israel/projects/new/lib/python3.5/site-packages/sphinx/ext/autodoc.py", line 658, in import_object
    __import__(self.modname)
  File "/home/israel/projects/fork/src/githubcity/ghcity.py", line 49, in <module>
    from githubcity.ghuser import GitHubUser
ImportError: No module named 'githubcity'
/home/israel/projects/fork/src/githubcity/ghregion.py:docstring of ghregion.GitHubRegion.addCity:4: WARNING: Field list ends without a blank line; unexpected unindent.
/home/israel/projects/fork/src/githubcity/ghregion.py:docstring of ghregion.GitHubRegion.export:6: WARNING: Definition list ends without a blank line; unexpected unindent.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] install                                                                                                         
generating indices... genindex py-modindex
writing additional pages... search
copying static files... WARNING: html_static_path entry '/home/israel/projects/fork/docs/_static' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 4 warnings.

Build finished. The HTML pages are in _build/html.
```

**Pull request closed without merging**